frontend/app/component/pollingstation/PollingStationFormNavigation.tsx

```diff
@@ -0,0 +1,206 @@
+    const currentSection = formState.sections[formState.current];
+    if (activeSection && currentSection) {
+      if (activeSection.index > currentSection.index) {
+        const url = getUrlForFormSection(currentSection.id);
+        navigate(url);
+      }
+    }
+    _lastKnownSection.current = formState.active;
+  }, [formState, navigate, getUrlForFormSection]);
+
+  //check if the targetFormSection has changed and navigate to the correct url
+  React.useEffect(() => {
+    if (!targetFormSection) return;
+    if (overrideControllerNavigation.current) {
+      const url = overrideControllerNavigation.current;
+      overrideControllerNavigation.current = null;
+      navigate(url);
+      return;
+    }
+    if (targetFormSection !== _lastKnownSection.current) {
+      _lastKnownSection.current = targetFormSection;
+      const url = getUrlForFormSection(targetFormSection);
+      navigate(url);
+    }
+  }, [targetFormSection, getUrlForFormSection, navigate]);
+
+  return (
+    <>
+      {blocker.state === "blocked" && (
+        <Modal
+          onClose={() => {
+            blocker.reset();
+          }}
+        >
+          <h2 id="modal-blocker-title">Let op: niet opgeslagen wijzigingen</h2>
+          <p>
+            Je hebt in{" "}
+            <strong>
+              {formState.sections[formState.active]?.title || "het huidige formulier"}
+            </strong>{" "}
+            wijzigingen gemaakt die nog niet zijn opgeslagen.
+          </p>
+          <p>Wil je deze wijzigingen bewaren?</p>
+          <nav>
+            <Button
+              size="lg"
+              onClick={() => {
+                overrideControllerNavigation.current = blocker.location.pathname;
+                submitCurrentForm();
+                blocker.reset();
+              }}
+            >
+              Wijzigingen opslaan
+            </Button>
+            <Button
+              size="lg"
+              variant="secondary"
+              onClick={() => {
+                blocker.proceed();
+              }}
+            >
+              Niet bewaren
+            </Button>
+          </nav>
+        </Modal>
+      )}
+
+      {error && (
+        <Feedback type="error" title="Server error">
+          <div id="feedback-server-error">
+            {error.errorCode}: {error.message}
+          </div>
+        </Feedback>
+      )}
+    </>
+  );
+}
+
+type BlockReason = "errors" | "warnings" | "changes";
+
+function reasonsBlocked(
+  formState: FormState,
+  currentForm: AnyFormReference,
+  values: PollingStationValues,
+): BlockReason[] {
+  const result: BlockReason[] = [];
+
+  const formSection = formState.sections[currentForm.id];
+  if (formSection) {
+    if (formSection.errors.length > 0) {
+      result.push("errors");
+    }
+    if (formSection.warnings.length > 0 && !formSection.ignoreWarnings) {
+      result.push("warnings");
+    }
+    //if (!formSection.isSubmitted) {
```

Comment:
This line (and line 202) can be removed?

Reply:
removed